irida_import/irida_import.py

```diff
@@ -628,6 +660,97 @@ def link(self, sample_file, folder_id):
 
         return added
 
+    def upload_file_to_galaxy(self, sample_file, folder_id):
+        """
+        Upload a sample file to Galaxy
+
+        :type sample_file: SampleFile
+        :param sample_file: the sample file to upload
+        :type folder_id: ID of folder to upload file to
+        :param sample_folder_path: the folder in Galaxy to store the file in
+        :return: a list containing a single dict with the file's
+        url, id, and name.
+        """
+        self.logger.debug('Attempting to upload file')
+        added = None
+        file_path = sample_file.path
+        self.logger.debug(
+            "       Sample file's local path is" + file_path)
+        file_type = 'auto'
+        file_ext = os.path.splitext(file_path)[1]
+
+        # Assume fastq files are fastqsanger:
+        if file_ext == '.fastq':
```

Comment:
Looking through IRIDA source code there is no place that enforces file extensions. Which means we need to rethink how we determine the file type. I also queried the database to check and we have some weird named files in there.

Reply:
I guess we were doing this before, so maybe it is acceptable here. We even have files that have spaces in there names...

Reply:
Yeah I just did it the way it was done before. Yup definitely need a better way to determine the type!